### PR TITLE
Move option parsing to libjulia

### PR DESCRIPTION
**Pull request by Keno**
For use in pyjulia and probably useful for other clients that are embedding julia as well. Since the options are in libjulia itself, I think it makes sense to put the option parsing in there as well.

The one option that is not strictly related to libjulia is `--lisp`. I have handled this by adding special code to repl.c to allow the common `julia --lisp`, but disallow `--lisp` in other positions since combining it with other arguments doesn't make much sense.

### CODE REVIEW

**Comment by vtjnash** [comment]
lgtm. looks like the test for `--lisp` needs to not test that all of the julia options get passed / handled correct however :)

File: src/jloptions.c [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+           opt_worker,
+           opt_bind_to,
+           opt_handle_signals,
+           opt_output_o,
+           opt_output_ji,
+           opt_use_precompiled,
+           opt_use_compilecache,
+           opt_incremental
+    };
+    static char* shortopts = "+vhqFfH:e:E:P:L:J:C:ip:O:";
```
Comment (vtjnash) [inline src/jloptions.c]: make this `const`

File: src/jloptions.c [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+    static struct option longopts[] = {
```
Comment (vtjnash) [inline src/jloptions.c]: and this

File: src/jloptions.c [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+        // exposed command line options
+        // NOTE: This set of required arguments need to be kept in sync
+        // with the required arguments defined in base/client.jl `process_options()`
```
Comment (tkelman) [inline src/jloptions.c]: there's a corresponding comment in `base/client.jl` that needs to be changed to reflect the move

File: src/julia.h [hunk 4]
```diff
@@ -1673,10 +1673,17 @@ typedef struct {
     const char *outputo;
     const char *outputji;
     int8_t incremental;
+    int8_t image_file_specified;
 } jl_options_t;
 
 extern JL_DLLEXPORT jl_options_t jl_options;
 
+// Parse an argc/argv pair to extract general julia options, passing back out
+// any arguments that should be passed on to the script. If imagepathspecifiedp
```
Comment (tkelman) [inline src/julia.h]: Not spelled `imagepathspecifiedp` any more?

**Pull request merged**